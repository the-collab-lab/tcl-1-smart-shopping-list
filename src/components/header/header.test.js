import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import Header from './index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot on load', () => {
  const tree = TestRenderer.create(
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

describe('different showBackLink values', () => {
  it('showBackLink === true', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Header showBackLink={true} />
      </BrowserRouter>
    );

    const headerChildren = tree.toJSON().children;
    expect(headerChildren.length).toBe(3);
    expect(headerChildren[0].props.visualstate).toBe('default');
    expect(headerChildren[1].props.visualstate).toBe('default');
    expect(headerChildren[2].props.visualstate).toBe('default');

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('showBackLink === false', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Header showBackLink={false} />
      </BrowserRouter>
    );

    const headerChildren = tree.toJSON().children;
    expect(headerChildren.length).toBe(3);
    expect(headerChildren[0].props.visualstate).toBe('hidden');
    expect(headerChildren[1].props.visualstate).toBe('default');
    expect(headerChildren[2].props.visualstate).toBe('default');

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('showBackLink not set', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    const headerChildren = tree.toJSON().children;
    expect(headerChildren.length).toBe(3);
    expect(headerChildren[0].props.visualstate).toBe('hidden');
    expect(headerChildren[1].props.visualstate).toBe('default');
    expect(headerChildren[2].props.visualstate).toBe('default');

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('different whichRoute values', () => {
  it('whichRoute === "/add-item"', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Header whichRoute="/add-item" />
      </BrowserRouter>
    );

    const headerChildren = tree.toJSON().children;
    expect(headerChildren.length).toBe(3);
    expect(headerChildren[0].props.href).toBe('/add-item');
    expect(headerChildren[1].props.href).toBe('/');
    expect(headerChildren[2].props.href).toBe('http://app.ineedtobuy.xyz/');

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('whichRoute === "/"', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Header whichRoute="/" />
      </BrowserRouter>
    );

    const headerChildren = tree.toJSON().children;
    expect(headerChildren.length).toBe(3);
    expect(headerChildren[0].props.href).toBe('/');
    expect(headerChildren[1].props.href).toBe('/');
    expect(headerChildren[2].props.href).toBe('http://app.ineedtobuy.xyz/');

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('whichRoute not set', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    const headerChildren = tree.toJSON().children;
    expect(headerChildren.length).toBe(3);
    expect(headerChildren[0].props.href).toBe('/');
    expect(headerChildren[1].props.href).toBe('/');
    expect(headerChildren[2].props.href).toBe('http://app.ineedtobuy.xyz/');

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
