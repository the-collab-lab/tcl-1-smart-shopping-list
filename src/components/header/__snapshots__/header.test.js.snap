// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`different showBackLink values showBackLink === false 1`] = `
<header
  className="css-527nkd"
>
  <a
    className="css-nuywz6"
    href="/"
    onClick={[Function]}
    visualstate="hidden"
  >
    Go Back
  </a>
  <a
    className="css-gq9b4g"
    href="/"
    onClick={[Function]}
    visualstate="default"
  >
    <h1
      className="css-r73csp"
    >
      iNeedToBuy
    </h1>
  </a>
  <a
    href="http://app.ineedtobuy.xyz/"
    target="_blank"
    visualstate="default"
  >
    See Example
  </a>
</header>
`;

exports[`different showBackLink values showBackLink === true 1`] = `
<header
  className="css-527nkd"
>
  <a
    className="css-gq9b4g"
    href="/"
    onClick={[Function]}
    visualstate="default"
  >
    Go Back
  </a>
  <a
    className="css-gq9b4g"
    href="/"
    onClick={[Function]}
    visualstate="default"
  >
    <h1
      className="css-r73csp"
    >
      iNeedToBuy
    </h1>
  </a>
  <a
    href="http://app.ineedtobuy.xyz/"
    target="_blank"
    visualstate="default"
  >
    See Example
  </a>
</header>
`;

exports[`different showBackLink values showBackLink not set 1`] = `
<header
  className="css-527nkd"
>
  <a
    className="css-nuywz6"
    href="/"
    onClick={[Function]}
    visualstate="hidden"
  >
    Go Back
  </a>
  <a
    className="css-gq9b4g"
    href="/"
    onClick={[Function]}
    visualstate="default"
  >
    <h1
      className="css-r73csp"
    >
      iNeedToBuy
    </h1>
  </a>
  <a
    href="http://app.ineedtobuy.xyz/"
    target="_blank"
    visualstate="default"
  >
    See Example
  </a>
</header>
`;

exports[`different whichRoute values whichRoute === "/" 1`] = `
<header
  className="css-527nkd"
>
  <a
    className="css-nuywz6"
    href="/"
    onClick={[Function]}
    visualstate="hidden"
  >
    Go Back
  </a>
  <a
    className="css-gq9b4g"
    href="/"
    onClick={[Function]}
    visualstate="default"
  >
    <h1
      className="css-r73csp"
    >
      iNeedToBuy
    </h1>
  </a>
  <a
    href="http://app.ineedtobuy.xyz/"
    target="_blank"
    visualstate="default"
  >
    See Example
  </a>
</header>
`;

exports[`different whichRoute values whichRoute === "/add-item" 1`] = `
<header
  className="css-527nkd"
>
  <a
    className="css-nuywz6"
    href="/add-item"
    onClick={[Function]}
    visualstate="hidden"
  >
    Go Back
  </a>
  <a
    className="css-gq9b4g"
    href="/"
    onClick={[Function]}
    visualstate="default"
  >
    <h1
      className="css-r73csp"
    >
      iNeedToBuy
    </h1>
  </a>
  <a
    href="http://app.ineedtobuy.xyz/"
    target="_blank"
    visualstate="default"
  >
    See Example
  </a>
</header>
`;

exports[`different whichRoute values whichRoute not set 1`] = `
<header
  className="css-527nkd"
>
  <a
    className="css-nuywz6"
    href="/"
    onClick={[Function]}
    visualstate="hidden"
  >
    Go Back
  </a>
  <a
    className="css-gq9b4g"
    href="/"
    onClick={[Function]}
    visualstate="default"
  >
    <h1
      className="css-r73csp"
    >
      iNeedToBuy
    </h1>
  </a>
  <a
    href="http://app.ineedtobuy.xyz/"
    target="_blank"
    visualstate="default"
  >
    See Example
  </a>
</header>
`;

exports[`snapshot on load 1`] = `
<header
  className="css-527nkd"
>
  <a
    className="css-nuywz6"
    href="/"
    onClick={[Function]}
    visualstate="hidden"
  >
    Go Back
  </a>
  <a
    className="css-gq9b4g"
    href="/"
    onClick={[Function]}
    visualstate="default"
  >
    <h1
      className="css-r73csp"
    >
      iNeedToBuy
    </h1>
  </a>
  <a
    href="http://app.ineedtobuy.xyz/"
    target="_blank"
    visualstate="default"
  >
    See Example
  </a>
</header>
`;
